src/mlpack/methods/ann/layer/linear.hpp

```diff
@@ -47,6 +47,13 @@ class Linear
    */
   Linear(const size_t inSize, const size_t outSize);
 
+  /**
+   * Create the Linear layer object using the specified layer size.
```

Comment:
It might be worth adding `the input layer size will automatically be inferred` to this comment, so that users understand the differences between this and the previous overload. :)